cips/cip-shwap_protocol.md

```diff
@@ -0,0 +1,378 @@
+---
+title: Shwap Protocol 
+description: Shwap - the new extensible sampling protocol 
+author: Hlib Kanunnikov (@Wondertan)
+discussions-to: https://forum.celestia.org/t/cip-shwap-protocol/1551
+status: Draft
+type: Standards Track
+category: Data Availability, Networking
+created: 2024-02-02
+---
+
+## Abstract
+
+This document specifies Shwap - the simple and expressive yet extensible and future-proof messaging framework aiming to
+solve critical inefficiencies and standardize messaging of Celestia's Data Availability p2p network.
+
+Shwap defines a messaging framework to be exchanged around the DA p2p network in a trust-minimized way without enforcing
+transport(QUIC/TCP or IP) or application layer protocol semantics(e.g., HTTP/x). Using this framework, Shwap
+declares the most common messages and provides options for stacking them with lower-level protocols.
+Shwap can be stacked together with application protocol like HTTP/x, [KadDHT][kaddht], [Bitswap][bitswap] or any custom
+protocol.
+
+## Motivation
+
+The current Data Availability Sampling (DAS) network protocol is inefficient. A _single_ sample operation takes log2(k)
+network roundtrips (where k is the square size). This is not practical and does not scale for the theoretically unlimited
+data square that the Celestia network enables. The main motive here is a protocol with O(1) roundtrip for _multiple_
+samples, preserving the assumption of having 1/n honest peers connected.
+
+Initially, Bitswap and IPLD were adopted as the basis for the DA network protocols, including DAS,
+block synchronization (BS), and blob/namespace data retrieval (ND). They gave battle-tested protocols and tooling with
+pluggability to rapidly scaffold Celestia's DA network. However, it came with the price of scalability limits and
+roundtrips, resulting in slower BS than block production. Before the network launch, the transition
+to the optimized [ShrEx protocol][shrex] for BS and integrating [CAR and DAGStore-based storage][storage] happened
+optimizing BS and ND. However, DAS was left untouched, preserving its weak scalability and roundtrip inefficiency.
```

Comment:
sentence feels incomplete or awkward between "happened" and "optimizing"

"for BS and integrating [CAR and DAGStore-based storage][storage] happened optimizing"

Reply:
Tried changing passive to active voice and it sounds better. WDYT?

> Before the network launch, we transitioned to the optimized [ShrEx protocol][shrex] for BS and integrated [CAR and DAGStore-based storage][storage] optimizing BS and ND.

Reply:
LGTM

[nit] do we need to clarify CAR and DAGStore acryonms?

Reply:
The link there clarifies them, so don't think so